components/Dialogs/src/MDCDialogTransitionController.m

```diff
@@ -18,16 +18,36 @@
 
 @implementation MDCDialogTransitionController
 
-static const NSTimeInterval MDCDialogTransitionDuration = 0.27;
+// The default duration of the dialog fade-in or fade-out animation
+static const NSTimeInterval kDefaultOpacityTransitionDuration = 0.2f;
+
+// The default duration of the dialog scale-up or scale-down animation
+static const NSTimeInterval kDefaultScaleTransitionDuration = 0.f;
+
+// The default starting X and Y scale of the presented dialog
+static const CGFloat kDefaultInitialScaleFactor = 1.0f;
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _opacityAnimationDuration = kDefaultOpacityTransitionDuration;
+    _scaleAnimationDuration = kDefaultScaleTransitionDuration;
+    _dialogInitialScaleFactor = kDefaultInitialScaleFactor;
+  }
+  return self;
+}
 
 #pragma mark - UIViewControllerAnimatedTransitioning
 
 - (NSTimeInterval)transitionDuration:
     (__unused id<UIViewControllerContextTransitioning>)transitionContext {
-  return MDCDialogTransitionDuration;
+  return MAX(self.opacityAnimationDuration, self.scaleAnimationDuration);
 }
 
 - (void)animateTransition:(id<UIViewControllerContextTransitioning>)transitionContext {
+  [CATransaction begin];
+  [CATransaction setAnimationDuration:[self transitionDuration:transitionContext]];
```

Comment:
Tip: Which animation is this duration controlling? It looks like all of the animations below have explicit durations.

Reply:
Initially, the duration of the full transition, but it's actually not necessary, so it will be removed.